Mage.Sets/src/mage/cards/v/VhalCandlekeepResearcher.java

```diff
@@ -0,0 +1,99 @@
+package mage.cards.v;
+
+import mage.ConditionalMana;
+import mage.MageInt;
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.Mana;
+import mage.abilities.SpellAbility;
+import mage.abilities.common.ChooseABackgroundAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.abilities.mana.ConditionalColorlessManaAbility;
+import mage.abilities.mana.builder.ConditionalManaBuilder;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.stack.Spell;
+
+import java.util.UUID;
+
+public final class VhalCandlekeepResearcher extends CardImpl {
+
+    public VhalCandlekeepResearcher(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{U}");
+        this.addSuperType(SuperType.LEGENDARY);
+
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(3);
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // {T}: Add an amount of {C} equal to Vhal, Candlekeep Researcher’s toughness. This mana can’t be spent to cast spells from your hand.
+        Ability ability = new ConditionalColorlessManaAbility(this.toughness.getValue(), new VhalCandlekeepResearcherManaBuilder());
+        ability.addCost(new TapSourceCost());
+
```

Comment:
The tap source cost is not necessary for abilities of ConditionalColorlessManaAbility type as they have a tap source cost by default, so line 43 can be removed. Additionally, you have to add this ability to the card using `this.addAbility(ability)`